### PR TITLE
Fix/online get

### LINKED ISSUE #1119: bug: if revision id is nil, online get/multi-get will panic
recurrent:

```shell
$ oomcli init
$ oomcli apply -f /tmp/tmp/meta.yaml
applied
$ oomcli get meta entity -o yaml
items:
  - kind: Entity
    name: device
    description: device
    groups:
      - name: phone
        category: batch
        description: phone
        features:
          - name: price
            value-type: int64
            description: price
          - name: model
            value-type: string
            description: model
  - kind: Entity
    name: user
    description: user
    groups:
      - name: student
        category: batch
        description: student
        features:
          - name: age
            value-type: int64
            description: age
      - name: user-click
        category: stream
        snapshot-interval: 1s
        description: user click post feature
        features:
          - name: last_5_click_posts
            value-type: string
            description: user last 5 click posts
          - name: number_of_user_started_posts
            value-type: int64
            description: number of posts that users stared today
$ oomcli get online -k 1 --feature phone.price
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x10e150c]

goroutine 1 [running]:
github.com/oom-ai/oomstore/internal/database/online/sqlutil.MultiGet({0x1da14e8, 0xc000120000}, 0x410f76, {{0xc000243690, 0x1, 0x1}, {0x1, {0xc00061b100, 0x5}, {0xc00061b130, ...}, ...}, ...}, ...)
        /home/lianxm/github.com/oomstore/internal/database/online/sqlutil/get.go:48 +0xec
github.com/oom-ai/oomstore/internal/database/online/sqlite.(*DB).MultiGet(0x20, {0x1da14e8, 0xc000120000}, {{0xc000243690, 0x1, 0x1}, {0x1, {0xc00061b100, 0x5}, {0xc00061b130, ...}, ...}, ...})
        /home/lianxm/github.com/oomstore/internal/database/online/sqlite/store.go:40 +0x6d
github.com/oom-ai/oomstore/pkg/oomstore.(*OomStore).getFeatureValueMap(0xc000163600, {0x1da14e8, 0xc000120000}, {0xc000243690, 0x1, 0x1}, 0x16e0e98, 0x1aa5ce6)
        /home/lianxm/github.com/oomstore/pkg/oomstore/online_query.go:111 +0x1e7
github.com/oom-ai/oomstore/pkg/oomstore.(*OomStore).OnlineMultiGet(0xc000163600, {0x1da14e8, 0xc000120000}, {{0xc0002436a0, 0x1, 0x1}, {0xc000243690, 0x1, 0x1}})
        /home/lianxm/github.com/oomstore/pkg/oomstore/online_query.go:84 +0x1c5
github.com/oom-ai/oomstore/oomcli/cmd.glob..func19(0x2b1b440, {0x1aa29c5, 0x4, 0x4})
        /home/lianxm/github.com/oomstore/oomcli/cmd/get_online.go:28 +0xf5
github.com/spf13/cobra.(*Command).execute(0x2b1b440, {0xc000162e80, 0x4, 0x4})
        /home/lianxm/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:860 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0x2b1cac0)
        /home/lianxm/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
        /home/lianxm/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/oom-ai/oomstore/oomcli/cmd.Execute()
        /home/lianxm/github.com/oomstore/oomcli/cmd/root.go:32 +0x25
main.main()
        /home/lianxm/github.com/oomstore/oomcli/main.go:6 +0x17
```

### PR DESCRIPTION
**Pull request by lianxmfor**
this PR does:
1. use *int rather than int for `GetOpt::RevisionID` and `MultiGetOpt::RevisionID`, and add enough comments
2. check nil in oomstore layer
3. check nil in internal db layer
4. online db add unit-test `TestNoRevisionID`

resolve #1119

### CODE REVIEW

**Comment by YikSanChan** [comment]
@lianxmfor Can you add some descriptions to the PR?

**Comment by lianxmfor** [comment]
> @lianxmfor Can you add some descriptions to the PR?

good advice! added a related issue.

File: internal/database/online/test_impl/get.go [hunk 1]
```diff
@@ -36,6 +38,43 @@ func TestGetExisted(t *testing.T, prepareStore PrepareStoreFn, destroyStore Dest
 	}
 }
 
+func TestGetNotRevision(t *testing.T, prepareStore PrepareStoreFn, destroyStore DestroyStoreFn) {
```
Comment (YikSanChan) [inline internal/database/online/test_impl/get.go:41]: How about `TestGetNilRevision`?

**Review by YikSanChan — COMMENTED** [review]
Can we check `revisionID == nil` earlier at oomstore layer? And make RevisionID `int` rather than `*int`

**Comment by lianxmfor** [comment]
> Can we check `revisionID == nil` earlier at oomstore layer? 

We can check if `revisionID == nil` at the oomstore level, but it is better to do it here also, which is in line with the principles of [defensive programming](https://en.wikipedia.org/wiki/Defensive_programming) and can avoid some problems.

> And make RevisionID `int` rather than `*int`

This is worth considering, one concern is that it is only needed if the feature is a `batch`, s it not a good idea to write `int` directly?

**Comment by lianxmfor** [comment]
Maybe we can change `RevisionID *int` to `BatchRevisionID int` ?

**Comment by YikSanChan** [comment]
> Maybe we can change `RevisionID *int` to `BatchRevisionID int` ?

That works for me, it is very clear

**Comment by lianxmfor** [comment]
> > Maybe we can change `RevisionID *int` to `BatchRevisionID int` ?
> 
> That works for me, it is very clear

done

**Review by YikSanChan — APPROVED** [review]
lgtm

**Comment by YikSanChan** [comment]
Let's:

- use `*int` rather than `int` for GetOpt::RevisionID, and add enough comments
- check nil in oomstore layer
- check nil in internal layer

**Comment by lianxmfor** [comment]
> Let's:
> 
> * use `*int` rather than `int` for GetOpt::RevisionID, and add enough comments
> * check nil in oomstore layer
> * check nil in internal layer

done

File: pkg/oomstore/online_query.go [hunk 2]
```diff
@@ -108,11 +114,17 @@ func (s *OomStore) getFeatureValueMap(ctx context.Context, entityKeys []string,
 		if len(features) == 0 {
 			continue
 		}
+
+		group := features[0].Group
+		if group.Category == types.CategoryBatch && group.OnlineRevisionID == nil {
+			return nil, errdefs.Errorf("the OnlineRevision of the group %s is nil,please check if there is no sync yet", group.Name)
```
Comment (YikSanChan) [inline pkg/oomstore/online_query.go:120]: group %s has nil OnlineRevision, please make sure you have already `sync`-ed

**Review by YikSanChan — APPROVED** [review]
lgtm

File: pkg/oomstore/online_query.go (hunk 2 above)
Comment (lianxmfor) [reply pkg/oomstore/online_query.go]: done

**Pull request merged**